cli/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+/// system. It configures, coordinates and manages transactions and queries processing, work of consensus and storage.
+pub struct Iroha<
+    W = World,
+    G = GenesisNetwork,
+    K = Kura<W>,
+    S = Sumeragi<G, K, W>,
+    B = BlockSynchronizer<S, W>,
+> where
+    W: WorldTrait,
+    G: GenesisNetworkTrait,
+    K: KuraTrait<World = W>,
+    S: SumeragiTrait<GenesisNetwork = G, Kura = K, World = W>,
+    B: BlockSynchronizerTrait<Sumeragi = S, World = W>,
+{
+    /// World state view
+    pub wsv: Arc<WorldStateView<W>>,
+    /// Queue of transactions
+    pub queue: Arc<Queue<W>>,
+    /// Sumeragi consensus
+    pub sumeragi: AlwaysAddr<S>,
+    /// Kura - block storage
+    pub kura: AlwaysAddr<K>,
+    /// Block synchronization actor
+    pub block_sync: AlwaysAddr<B>,
+    /// Torii web server
+    pub torii: Option<Torii<W>>,
+}
+
+impl<W, G, S, K, B> Iroha<W, G, K, S, B>
+where
+    W: WorldTrait,
+    G: GenesisNetworkTrait,
+    K: KuraTrait<World = W>,
+    S: SumeragiTrait<GenesisNetwork = G, Kura = K, World = W>,
+    B: BlockSynchronizerTrait<Sumeragi = S, World = W>,
+{
+    /// To make `Iroha` peer work all actors should be started first.
+    /// After that moment it you can start it with listening to torii events.
+    ///
+    /// # Errors
+    /// - Reading genesis from disk
+    /// - Reading telemetry configs
+    /// - telemetry setup
+    /// - Initialisation of [`Sumeragi`]
+    pub async fn new(
+        args: &Arguments,
+        instruction_validator: IsInstructionAllowedBoxed<K::World>,
+        query_validator: IsQueryAllowedBoxed<K::World>,
+    ) -> Result<Self> {
+        let broker = Broker::new();
+        let mut config = Configuration::from_path(&args.config_path)?;
+        config.load_environment()?;
+
+        let genesis = G::from_configuration(
+            args.submit_genesis,
+            RawGenesisBlock::from_path(&args.genesis_path)?,
+            &config.genesis,
+            &config.sumeragi.transaction_limits,
+        )
+        .wrap_err("Failed to initialize genesis.")?;
+
+        Self::with_genesis(
+            genesis,
+            config,
+            instruction_validator,
+            query_validator,
+            broker,
+        )
+        .await
+    }
+
+    /// Create Iroha with specified broker, config, and genesis.
+    ///
+    /// # Errors
+    /// - Reading telemetry configs
+    /// - telemetry setup
+    /// - Initialization of [`Sumeragi`]
+    #[allow(clippy::non_ascii_literal)]
+    pub async fn with_genesis(
+        genesis: Option<G>,
+        config: Configuration,
+        instruction_validator: IsInstructionAllowedBoxed<K::World>,
+        query_validator: IsQueryAllowedBoxed<K::World>,
+        broker: Broker,
+    ) -> Result<Self> {
+        if !config.disable_panic_terminal_colors {
+            color_eyre::install()?;
+        }
+        let telemetry = iroha_logger::init(&config.logger)?;
+        iroha_logger::info!("Hyperledgerいろは2にようこそ！");
```

Comment:
is everything ok with these whitespace characters. They look like squares on my browser

Reply:
![image](https://user-images.githubusercontent.com/49983831/158789685-4bccf40d-57b2-48c4-83c7-42709ec7d518.png)
Saying "Welcome to Hyperledger Iroha 2!".
I was wondering why only here is Japanese. Should be English

Reply:
oh, I think that's fine then. I'd leave it in Japanese, I like that Iroha includes lots of references to Japanese culture.

Reply:
It's a minimal flourish that adds lots of flavour. I'll add the translation, but keep the japanese version.